packages/app/src/systems/Settings/pages/ChangePassword.tsx

```diff
@@ -0,0 +1,141 @@
+import { cssObj } from '@fuel-ui/css';
+import { Alert, Button, InputPassword } from '@fuel-ui/react';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+import * as yup from 'yup';
+
+import { useSettings } from '../hooks/useSettings';
+
+import { ControlledField, Layout, Pages } from '~/systems/Core';
+
+const schema = yup
+  .object({
+    newPassword: yup
+      .string()
+      .min(8, 'Your new password must to have at least 8 characters')
+      .required('New Password is required'),
+    confirmPassword: yup
+      .string()
+      .oneOf([yup.ref('newPassword'), null], 'Passwords must match'),
+    oldPassword: yup.string().required('Old Password is required'),
+  })
+  .required();
+
+type ChangePasswordFormValues = {
+  confirmPassword: string;
+  newPassword: string;
+  oldPassword: string;
+};
+
+export function ChangePassword() {
+  const navigate = useNavigate();
+  const { changePassword, isChangingPassword, hasChangedPassword } =
+    useSettings();
+  const { handleSubmit, control } = useForm<ChangePasswordFormValues>({
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+    resolver: yupResolver(schema),
+  });
+
+  function onSubmit({ newPassword, oldPassword }: ChangePasswordFormValues) {
+    changePassword({
+      oldPassword,
+      newPassword,
+    });
+  }
+
+  useEffect(() => {
+    if (hasChangedPassword) {
+      navigate(Pages.wallet());
```

Comment:
This should be inside a machine action, not a `useEffect`. Always try to avoid `useEffect` when possible.